EducationPortal/EducationPortal.ConsoleUI/Validation/MaterialDataValidator.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+using EducationPortal.BLL;
+using EducationPortal.BLL.DTO;
+
+namespace EducationPortal.ConsoleUI.Validation
+{
+    public class MaterialDataValidator
+    {
+        private MaterialDTO material;
+        private Uri uri;
+
+        public MaterialDataValidator(MaterialDTO material)
+        {
+            this.material = material;
+        }
+
+        public ValidationResult Validate()
+        {
+            if (material.Name.Length < DataSettings.MaterialNameMinCharacterCount
+             || material.Name.Length > DataSettings.MaterialNameMaxCharacterCount)
+            {
+                return new ValidationResult()
+                {
+                    IsValid = false,
+                    Message = string.Format("Название материала должно быть длиной от {0} до {1} символов!",
+                                             DataSettings.MaterialNameMinCharacterCount,
+                                             DataSettings.MaterialNameMaxCharacterCount)
+                };
+
+            }
+
+            var isUrlCorrect = Uri.TryCreate(material.Url, UriKind.Absolute, out uri)
+                            && (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps);
+
+            if (!isUrlCorrect)
+            {
+                return new ValidationResult()
+                {
+                    IsValid = false,
+                    Message = "Некорректный формат URL!"
+                };
+            }
+
+            if (material is BookDTO)
```

Comment:
Maybe you should try to use a base validator and override it on specific classes